### PR TITLE
Update mobile react 18 types/usages and various packages

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description
- Updates react and react-native types
- Fixes type issues
- Upgrades react-native libraries that have fixed types
- Removes react-native-shadow2 in place of homegrown shadow component
- Patches old markdown library with type shim
- fixes react-native-video
- fixes potential issue with @react-native-firebase

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-update-react18-types-usage

**Review by raymondjacobson — APPROVED** [review]
Changes look good. hand rolled shadow seems a bit scary to me because we have multiple overlapping ones often, but down to play with it on staging, esp if it improves perf

File: packages/mobile/src/components/core/Shadow.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import { useMemo } from 'react'
+
+import type { StyleProp, ViewStyle } from 'react-native'
+import { View } from 'react-native'
+
+import { makeStyles, shadow } from 'app/styles'
+
+type ShadowConfig = Pick<
+  ViewStyle,
+  'shadowOffset' | 'shadowColor' | 'shadowOpacity' | 'shadowRadius'
```
[the remark this answers is not in the document]
Comment (raymondjacobson) [reply packages/mobile/src/components/core/Shadow.tsx:11]: > actually. won't this only work on ios?

https://reactnative.dev/docs/shadow-props

[the remark this answers is not in the document]
Comment (dylanjeffers) [reply packages/mobile/src/components/core/Shadow.tsx:11]: yes but we have the elevation key to handle android. but if we feel that would look too weird we can change. we are using these styles for tiles currently on android

**Pull request merged**